controllers/virtualmachinesnapshot/virtualmachinesnapshot_controller.go

```diff
@@ -319,6 +297,105 @@ func ensureCSIVolumeSyncAnnotation(vmSnapshot *vmopv1.VirtualMachineSnapshot) {
 	}
 	// As long as the value is not set to completed by CSI driver, we need to mark it as requested.
 	if vmSnapshot.Annotations[constants.CSIVSphereVolumeSyncAnnotationKey] != constants.CSIVSphereVolumeSyncAnnotationValueCompleted {
-		vmSnapshot.Annotations[constants.CSIVSphereVolumeSyncAnnotationKey] = constants.CSIVSphereVolumeSyncAnnotationValueRequest
+		vmSnapshot.Annotations[constants.CSIVSphereVolumeSyncAnnotationKey] = constants.CSIVSphereVolumeSyncAnnotationValueRequested
+	}
+}
+
+func setSnapshotReadyCondition(vmSnapshot *vmopv1.VirtualMachineSnapshot) {
+	// If the snapshot is being deleted, skip setting the condition.
+	if !vmSnapshot.DeletionTimestamp.IsZero() {
+		return
 	}
+
+	if pkgcnd.Get(vmSnapshot, vmopv1.VirtualMachineSnapshotCreatedCondition) == nil {
```

Comment:
Hmm.  Since we are manipulating the conditions on the snapshot resource from the VM provider code anyway, would the following workflow be simpler:
- Set the "SnapshotCreationInProgress" from the VM provider when the snapshot is being processed.  We don't set any conditions before then
- If creation fails, the VM provider sets the created to fails with failed reason
- If creation succeeds, then the snapshot controller takes over by applying the CSI annotation and setting the CSI sync condition.  We can also use the created condition to update the storage (like you are already doing).
Thoughts?

Reply:
Ok, I read from somewhere it's a good practice to always have a condition. Let me change that